core/src/test/java/nl/nn/adapterframework/util/CompactSaxHandlerTest.java

```diff
@@ -174,6 +175,22 @@ void testElementToMoveChain() throws IOException, SAXException {
 		assertEquals("DC2022-012345", session.getString("ref_identificatie2"));
 	}
 
+	@Test
+	void testElementToMoveChainChomped() throws IOException, SAXException {
```

Comment:
It seems to me that this behaviour is wrong. We could move the element, but instead we chomp it and truncate data.

If that was old behaviour, I'm not sure if it really was behaviour that anyone wanted.

Reply:
Yup, that's why it's handy and important to test the class first before refactoring.

Reply:
I believe the original class did chomp the elements that were moved, and that changing this is what made the larva tests fail.

Reply:
The current implementation does not move elements that are chomped. And we did add tests before refactoring and covered all logic lines. But this is an odd combination/scenario, which happened to came out of the Larva tests. I don't like late notifications, so I've added this tests to unit tests.

Reply:
I do indeed prefer this to be tested by unit tests, but it's behaviour that does not make sense to me.

If you want to move an element, you do so (in my view) to preserve its data and use it outside the XML. So an element-to-move is important to you, even if it's large.

Other large elements might not be so relevant to your process, so you might be happy to chomp them.

That's why I personally prefer the behaviour as we had it before the weekend, even if that broke Larva tests -- but the Larva tests tell is we are making a chance in behaviour so we have to make it consciously and explicitly, and also update the Larva tests. (Or perhaps we can remove them if they fully overlap with new unit tests? @nielsm5 would you agree with removing some Larva tests if the behaviour is now fully covered by unit tests? And the tests are not testing any kind of system integration?

Reply:
Ok, to wrap up, I think @tnleeuw 's idea sounds good. I suggest the following, up for both @nielsm5 and @tnleeuw approval:

1. Do NOT do a size check (chomping) when we're processing the 'element to move' (same for move chain). So, I'll revert the single added line to the CompactSaxHandler class, I've added to 'fix' it.
2. Make unit tests compatible with the above statement.
3. Remove the Larva tests who failed, with the wrong chomping check. Still keep a few Larva tests, to test the basic behavior, because it is used by the Receiver etc, which is a good idea to keep (integrated) tested. 

Both agree? I will wait for both approvals before doing anything (and revert it later).

Reply:
Sounds good to me.

Reply:
As discussed and agreed today at the office, we make it like described in the 3 points above.